Controller/CRUDController.php

```diff
@@ -511,7 +511,16 @@ protected function redirectTo($object, Request $request = null)
         }
 
         if (!$url) {
-            $url = $this->admin->generateObjectUrl('edit', $object);
+            foreach (array('edit', 'show') as $route) {
+                if ($this->admin->hasRoute($route) && $this->admin->isGranted(strtoupper($route))) {
+                    $url = $this->admin->generateObjectUrl($route, $object);
+                    break;
```

Comment:
In any case, this method return a `RedirectResponse` at the end. So we could return it directly instead continue onto the code process.